acme/challenge.go

```diff
@@ -446,158 +441,178 @@ func wireOIDC01Validate(ctx context.Context, ch *Challenge, db DB, jwk *jose.JSO
+type wireDpopPayload struct {
+	// AccessToken is the token generated by wire-server
+	AccessToken string `json:"access_token,omitempty"`
+}
+
+func wireDPOP01Validate(ctx context.Context, ch *Challenge, db DB, accountJWK *jose.JSONWebKey, payload []byte) error {
 	prov, ok := ProvisionerFromContext(ctx)
 	if !ok {
 		return NewErrorISE("missing provisioner")
 	}
 
-	rawKid, err := jwk.Thumbprint(crypto.SHA256)
-	if err != nil {
-		return storeError(ctx, db, ch, false, WrapError(ErrorServerInternalType, err, "failed to compute JWK thumbprint"))
-	}
-	kid := base64.RawURLEncoding.EncodeToString(rawKid)
-
-	dpopOptions := prov.GetOptions().GetDPOPOptions()
-	key := dpopOptions.GetSigningKey()
-
-	var wireChallengePayload WireChallengePayload
-	if err := json.Unmarshal(payload, &wireChallengePayload); err != nil {
+	var dpopPayload wireDpopPayload
+	if err := json.Unmarshal(payload, &dpopPayload); err != nil {
 		return storeError(ctx, db, ch, false, WrapError(ErrorRejectedIdentifierType, err,
 			"error unmarshalling Wire challenge payload"))
 	}
 
-	file, err := os.CreateTemp(os.TempDir(), "acme-validate-challenge-pubkey-")
-	if err != nil {
-		return WrapErrorISE(err, "temporary file could not be created")
-	}
-	defer file.Close()
-	defer os.Remove(file.Name())
-
-	buf := bytes.NewBuffer(nil)
-	buf.WriteString(key)
-
-	n, err := file.Write(buf.Bytes())
-	if err != nil {
-		return WrapErrorISE(err, "failed writing signature key to temp file")
-	}
-	if n != buf.Len() {
-		return WrapErrorISE(err, "expected to write %d characters to the key file, got %d", buf.Len(), n)
-	}
-
-	challengeValues, err := wire.ParseID([]byte(ch.Value))
+	wireID, err := wire.ParseID([]byte(ch.Value))
 	if err != nil {
 		return WrapErrorISE(err, "error unmarshalling challenge data")
 	}
 
-	clientID, err := wire.ParseClientID(challengeValues.ClientID)
+	clientID, err := wire.ParseClientID(wireID.ClientID)
 	if err != nil {
 		return WrapErrorISE(err, "error parsing device id")
 	}
 
+	dpopOptions := prov.GetOptions().GetWireOptions().GetDPOPOptions()
 	issuer, err := dpopOptions.GetTarget(clientID.DeviceID)
 	if err != nil {
 		return WrapErrorISE(err, "invalid Go template registered for 'target'")
 	}
 
-	expiry := strconv.FormatInt(time.Now().Add(time.Hour*24*365).Unix(), 10)
-	cmd := exec.CommandContext( //nolint:gosec // TODO(hs): replace this with Go implementation
-		ctx,
-		dpopOptions.GetValidationExecPath(),
-		"verify-access",
-		"--client-id",
-		challengeValues.ClientID,
-		"--handle",
-		challengeValues.Handle,
-		"--challenge",
-		ch.Token,
-		"--leeway",
-		"360",
-		"--max-expiry",
-		expiry,
-		"--issuer",
-		issuer,
-		"--hash-algorithm",
-		`SHA-256`,
-		"--kid",
-		kid,
-		"--key",
-		file.Name(),
-	)
-
-	stdin, err := cmd.StdinPipe()
+	params := verifyParams{
+		token:      dpopPayload.AccessToken,
+		key:        dpopOptions.GetSigningKey(),
+		accountJWK: accountJWK,
```

Comment:
I commented on the other PR something about this, about the names, and that you can just pass the public key, accountJWK.Key